status/reporters.go

```diff
@@ -413,3 +551,146 @@ func (r *GitHubReporter) ReportStatus(k8sClient client.Client, ctx context.Conte
 
 	return nil
 }
+
+// ReportStatusForSnapshot creates CheckRun when using GitHub App integration,
+// creates a a commit status when using GitHub webhook integration
+func (r *GitHubReporter) ReportStatusForSnapshot(k8sClient client.Client, ctx context.Context, logger *helpers.IntegrationLogger, snapshot *applicationapiv1alpha1.Snapshot) error {
+	if !metadata.HasLabelWithValue(snapshot, gitops.PipelineAsCodeEventTypeLabel, gitops.PipelineAsCodePullRequestType) {
```

Comment:
Wondering if we should have this check as part of the predicate to prevent starting a reconciliation loop for Snapshots that aren't of the `pull_request` type?

Reply:
Just a reminder to update the controller diagram too.

Reply:
Updated. Thanks!